### ISSUE TITLE
Autogenerated category with directory prevents build

### ISSUE THREAD

**Issue by andrewnicols** [facebook/docusaurus#7358]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#reporting-new-issues).

### Prerequisites

- [X] I'm using the latest version of Docusaurus.
- [X] I have tried the `npm run clear` or `yarn clear` command.
- [X] I have tried `rm -rf node_modules yarn.lock package-lock.json` and re-installing packages.
- [X] I have tried creating a repro with https://new.docusaurus.io.
- [X] I have read the console error message carefully (if applicable).

### Description

Given the following directory structure:
```
.
└── docs
    └── tutorials
        └── tutorails-basics
            └── index.md
```

And the following `sidebars.js` configuration:
```javascript
const sidebars = {
  docs: [
    'intro',
    {
      label: 'Tutorials',
      type: 'category',
      items: [
        {. 
          type: 'autogenerated',
          dirName: 'tutorials',
        },
      ],
    },
  ],
};
```

Then I should see a nested sidebar

However, I get the following exception:
```
❯ yarn start
$ docusaurus start
[INFO] Starting the development server...
[ERROR] Sidebars file at "/home/projects/github-1awfyk/sidebars.js" failed to be loaded.
[ERROR] Loading of version failed for version current
[ERROR] Error: Sidebar category Tutorials has neither any subitem nor a link. This makes this item not able to link to anything.
    at postProcessSidebarItem (/home/projects/github-1awfyk/node_modules/@docusaurus/plugin-content-docs/lib/sidebars/postProcessor.js:40:23)
    at eval (/home/projects/github-1awfyk/node_modules/@docusaurus/plugin-content-docs/lib/sidebars/postProcessor.js:63:84)
    at Array.map (<anonymous>)
    at eval (/home/projects/github-1awfyk/node_modules/@docusaurus/plugin-content-docs/lib/sidebars/postProcessor.js:63:70)
    at eval (/home/projects/github-1awfyk/node_modules/lodash/lodash.js:13469:38)
    at eval (/home/projects/github-1awfyk/node_modules/lodash/lodash.js:4967:15)
    at baseForOwn (/home/projects/github-1awfyk/node_modules/lodash/lodash.js:3032:24)
    at Function.mapValues (/home/projects/github-1awfyk/node_modules/lodash/lodash.js:13468:7)
    at postProcessSidebars (/home/projects/github-1awfyk/node_modules/@docusaurus/plugin-content-docs/lib/sidebars/postProcessor.js:63:29)
    at loadSidebars (/home/projects/github-1awfyk/node_modules/@docusaurus/plugin-content-docs/lib/sidebars/index.js:85:56)
    at async doLoadVersion (/home/projects/github-1awfyk/node_modules/@docusaurus/plugin-content-docs/lib/index.js:93:34)
    at async loadVersion (/home/projects/github-1awfyk/node_modules/@docusaurus/plugin-content-docs/lib/index.js:115:28)
    at async Promise.all (index 0)
    at async Object.loadContent (/home/projects/github-1awfyk/node_modules/@docusaurus/plugin-content-docs/lib/index.js:123:33)
    at async eval (/home/projects/github-1awfyk/node_modules/@docusaurus/core/lib/server/plugins/index.js:35:25)
    at async Promise.all (index 0)
```

However, having a directory which only contains child directories works if the autogenerated directory is at the top:
```
const sidebars = {
  tutorialSidebar: [{type: 'autogenerated', dirName: '.'}],
};
```

### Reproducible demo

https://stackblitz.com/edit/github-1awfyk?file=sidebars.js

### Steps to reproduce

Create a sidebar configuration where the directory described in an autogenerated sidebar only has _directories_ *directly* inside it.

### Expected behavior

Nested auto-generated indexes should work from any level, as they did in beta 18.

### Actual behavior

```
❯ yarn start
$ docusaurus start
[INFO] Starting the development server...
[ERROR] Sidebars file at "/home/projects/github-1awfyk/sidebars.js" failed to be loaded.
[ERROR] Loading of version failed for version current
[ERROR] Error: Sidebar category Tutorials has neither any subitem nor a link. This makes this item not able to link to anything.
    at postProcessSidebarItem (/home/projects/github-1awfyk/node_modules/@docusaurus/plugin-content-docs/lib/sidebars/postProcessor.js:40:23)
    at eval (/home/projects/github-1awfyk/node_modules/@docusaurus/plugin-content-docs/lib/sidebars/postProcessor.js:63:84)
    at Array.map (<anonymous>)
    at eval (/home/projects/github-1awfyk/node_modules/@docusaurus/plugin-content-docs/lib/sidebars/postProcessor.js:63:70)
    at eval (/home/projects/github-1awfyk/node_modules/lodash/lodash.js:13469:38)
    at eval (/home/projects/github-1awfyk/node_modules/lodash/lodash.js:4967:15)
    at baseForOwn (/home/projects/github-1awfyk/node_modules/lodash/lodash.js:3032:24)
    at Function.mapValues (/home/projects/github-1awfyk/node_modules/lodash/lodash.js:13468:7)
    at postProcessSidebars (/home/projects/github-1awfyk/node_modules/@docusaurus/plugin-content-docs/lib/sidebars/postProcessor.js:63:29)
    at loadSidebars (/home/projects/github-1awfyk/node_modules/@docusaurus/plugin-content-docs/lib/sidebars/index.js:85:56)
    at async doLoadVersion (/home/projects/github-1awfyk/node_modules/@docusaurus/plugin-content-docs/lib/index.js:93:34)
    at async loadVersion (/home/projects/github-1awfyk/node_modules/@docusaurus/plugin-content-docs/lib/index.js:115:28)
    at async Promise.all (index 0)
    at async Object.loadContent (/home/projects/github-1awfyk/node_modules/@docusaurus/plugin-content-docs/lib/index.js:123:33)
    at async eval (/home/projects/github-1awfyk/node_modules/@docusaurus/core/lib/server/plugins/index.js:35:25)
    at async Promise.all (index 0)
```

### Your environment

- Public source code: https://github.com/moodle/devdocs
- Public site URL: https://moodle.github.io/devdocs
- Docusaurus version used: beta.18, but this issue is in beta.20
- Environment name and version Node v16.15.0
- Operating system and version (e.g. Ubuntu 20.04.2 LTS): MacOS, Stackblitz

### Self-service

- [ ] I'd be willing to fix this bug myself.

**Comment by Josh-Cena** [comment]
Duplicate of #7348

**Comment by Josh-Cena** [comment]
Fixed in #7385. Please try the [canary version](https://docusaurus.io/community/canary).
